internal/serve/httphandler/circle_config_handler_test.go

```diff
@@ -0,0 +1,128 @@
+	}
+	r := chi.NewRouter()
+	u := "/organization/circle-config"
+	r.Patch(u, handler.Patch)
+
+	validPatchRequest := PatchCircleConfigRequest{
+		ApiKey:   utils.StringPtr("new_api_key"),
+		WalletID: utils.StringPtr("new_wallet_id"),
+	}
+
+	mCircleDistributionAccount := mDistributionAccountResolver.
+		On("DistributionAccountFromContext", mock.Anything).
+		Return(schema.TransactionAccount{Type: schema.DistributionAccountCircleDBVault}, nil)
+
+	t.Run("returns bad request for invalid request body", func(t *testing.T) {
+		mCircleDistributionAccount.Once()
+
+		rr := request(t, r, http.MethodPatch, strings.NewReader("invalid json"))
+
+		assert.Equal(t, http.StatusBadRequest, rr.Code)
+		assert.JSONEq(t, `{"error": "Request body is not valid"}`, rr.Body.String())
+	})
+
+	t.Run("returns bad request for invalid patch request", func(t *testing.T) {
+		mCircleDistributionAccount.Once()
+
+		invalidPatchRequest := PatchCircleConfigRequest{}
+		body, err := json.Marshal(invalidPatchRequest)
+		require.NoError(t, err)
+
+		rr := request(t, r, http.MethodPatch, strings.NewReader(string(body)))
+
+		assert.Equal(t, http.StatusBadRequest, rr.Code)
+		assert.JSONEq(t, `{"error":"Request body is not valid", "extras":{"validation_error":"wallet_id or api_key must be provided"}}`, rr.Body.String())
+	})
+
+	t.Run("returns bad request if distribution account type is not Circle", func(t *testing.T) {
+		mDistributionAccountResolver.
+			On("DistributionAccountFromContext", mock.Anything).
+			Return(schema.TransactionAccount{Type: schema.DistributionAccountStellarEnv}, nil).
+			Once()
+
+		body, err := json.Marshal(validPatchRequest)
+		require.NoError(t, err)
+
+		rr := request(t, r, http.MethodPatch, strings.NewReader(string(body)))
+
+		assert.Equal(t, http.StatusBadRequest, rr.Code)
+		assert.JSONEq(t, `{"error": "This endpoint is only available for tenants using CIRCLE"}`, rr.Body.String())
+	})
+
+	t.Run("updates Circle configuration successfully", func(t *testing.T) {
+		mCircleDistributionAccount.Once()
+
+		body, err := json.Marshal(validPatchRequest)
+		require.NoError(t, err)
+
+		rr := request(t, r, http.MethodPatch, strings.NewReader(string(body)))
+		assert.Equal(t, http.StatusOK, rr.Code)
+
+		// Check the updated config in the database
+		config, err := ccm.Get(context.Background())
+		require.NoError(t, err)
+		require.NotNil(t, config)
+		assert.Equal(t, "new_wallet_id", *config.WalletID)
+
+		decryptedApiKey, err := encrypter.Decrypt(*config.EncryptedApiKey, encryptionPassphrase)
+		assert.NoError(t, err)
+		assert.Equal(t, "new_api_key", decryptedApiKey)
+		assert.Equal(t, encryptionPublicKey, *config.EncrypterPublicKey)
+	})
+}
+
+func request(t *testing.T, r *chi.Mux, httpMethod string, body io.Reader) *httptest.ResponseRecorder {
+	t.Helper()
+
+	req, err := http.NewRequest(httpMethod, "/organization/circle-config", body)
+	require.NoError(t, err)
+
+	rr := httptest.NewRecorder()
+	r.ServeHTTP(rr, req)
+	return rr
+}
```

Comment:
It'd be nice to eventually move this to a package where other tests can import as well. Not something you need to do on thie PR though.

Reply:
addressed in 7a1753216fcf05918848edf80df7275c7527ddad